engine/src/main/java/org/hibernate/search/engine/search/timeout/spi/TimeoutManager.java

```diff
@@ -79,11 +80,13 @@ public TimeoutManager(TimingSource timingSource, Long timeoutValue, TimeUnit tim
 	 * we start counting from this method call (if needed)
 	 */
 	public void start() {
-		this.start = timingSource.monotonicTimeEstimate();
+		this.monotonicTimeEstimateStart = timingSource.monotonicTimeEstimate();
```

Comment:
Actually, if no timeout was defined, this line becomes unnecessary.

That in itself wouldn't be worth discussing, but... if you only execute this line when there actually is a timeout, then you can do the same for `timingSource.ensureInitialized();`, in the constructor above. And *that* method can be costly, as it triggers the creation of a thread pool on its first execution.

Reply:
Right, we don't need it anymore to compute the took time!